### PR TITLE
adding ability to have teams with cluster access

### PR DESCRIPTION
**Pull request by sinamna**

### CODE REVIEW

**Review by SadeghTb — CHANGES_REQUESTED** [review]

File: controllers/namespace_controller.go [hunk 1]
```diff
@@ -330,7 +335,11 @@ func (r *NamespaceReconciler) createAppProj(team string) (*argov1alpha1.AppProje
 			},
 		},
 	}
-
+	if isTeamClusterScoped(team, team_list) {
+		appProj.Spec.ClusterResourceWhitelist = includeAllGroupKind
```
[the remark this answers is not in the document]
Comment (sinamna) [reply controllers/namespace_controller.go:339]: we explicitly should set whitelist

File: controllers/namespace_controller.go [hunk 2]
```diff
@@ -292,6 +292,17 @@ func (r *NamespaceReconciler) createAppProj(team string) (*argov1alpha1.AppProje
 	repo_env := os.Getenv("PUBLIC_REPOS")
 	repo_list := strings.Split(repo_env, ",")
 
+	// Get cluster scoped teams
+	team_env := os.Getenv("TEAMS")
```
[the remark this answers is not in the document]
Comment (sinamna) [reply controllers/namespace_controller.go:296]: I'll Change the naming. its not all teams, just the ones we one to whitelist their access

File: config/manager/manager.yaml [hunk 3]
```diff
@@ -55,5 +55,7 @@ spec:
         envFrom:
         - configMapRef:
              name: public-repos
+        - configMapRef:
+             name: cluster-scoped-teams
```
[the remark this answers is not in the document]
Comment (sinamna) [reply config/manager/manager.yaml]: updated to cluster-admin-teams

File: controllers/namespace_controller.go [hunk 4]
```diff
@@ -371,3 +380,12 @@ func convertLabelToAppProjectNameset(l string) AppProjectNameset {
 	}
 	return result
 }
+
+func isTeamClusterScoped(team string, clusterScopedList []string) bool {
```
[the remark this answers is not in the document]
Comment (sinamna) [reply controllers/namespace_controller.go]: cluster admin teams are seperated by "," in config. All teams trigger the reconcile logic and we should seperate admin ones by checking the list provided by config

**Pull request merged**